analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -56,8 +56,11 @@ public class SymbolicExecutionRunner : SymbolicExecutionRunnerBase
             new InitializationVectorShouldBeRandom(),
             new HashesShouldHaveUnpredictableSalt());
 
-        protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules => ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
-            .Add(LocksReleasedAllPaths.S2222, CreateFactory<LocksReleasedAllPaths>());
+        protected override ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules { get; } = new Dictionary<DiagnosticDescriptor, RuleFactory>
+        {
+            { RuleChecks.LocksReleasedAllPaths.S2222, CreateFactory<RuleChecks.LocksReleasedAllPaths>() },
+            { RuleChecks.NullPointerDereference.S2259, CreateFactory<RuleChecks.NullPointerDereference>() },
+        }.ToImmutableDictionary();
```

Comment:
I changed it to a read-only property with an initializer because `AllRules` is iterated at least twice in SymbolicExecutionRunnerBase. This way, the dictionary is cached in a field instead of created on each access.
I think, the dictionary initializer with the trailing commas is easier maintainable (copy line) and should be slightly better performing allocation-wise.

Reply:
The readonly property is fine. But the collection initializer and `.ToImmutableDictionary()` is additional complication.
Please keep the readonly property, but return the immutable buildering back to avoid the noisy scaffolding.